app.js

```diff
@@ -137,38 +138,167 @@ class Products {
+  }
+  addCartItems(item) {
+    const div = document.createElement("div");
+    div.classList.add("cart-item");
+    div.innerHTML = `<img src=${item.image} alt="product" />
+        <div class="text-black">
+          <h4 class="capitalize">${item.title}</h4>
+          <h5>$${item.price}</h5>
+          <span class="remove-item cursor-pointer" data-id = ${item.id}>remove</span>
         </div>
-        `; 
-      cartContent.appendChild(div);
-    }
+        <div class="text-black"> 
+        <i class="fas fa-chevron-up cursor-pointer" data-id= ${item.id}></i>
+        <p class="item-amount self-center">${item.amount}</p>
+        <i class="fas fa-chevron-down cursor-pointer" data-id= ${item.id}></i>
+      </div>
+      `; 
+    cartContent.appendChild(div);
+  }
+  showCart() {
+    cartOverlay.classList.add("transparentBcg");
+    cartDom.classList.add("showCart");
+  }
+  //a method upon the loading of the page, to check the cart value from the local storage.
+  setupAPP() {
+    cart = Storage.getCart();
+    this.setCartValues(cart);
+    this.populateCart(cart);
+    //EventListeners for Displaying and Hiding the cart.
+    cartBtn.addEventListener("click", this.showCart);
+    closeCartBtn.addEventListener("click", this.hideCart);
+  }
+  populateCart(cart) {
+    cart.forEach((item) => this.addCartItems(item));
+  }
+  //to close cart
+  hideCart() {
+    cartOverlay.classList.remove("transparentBcg");
+    cartDom.classList.remove("showCart");
+  }
+
+  //A method for the cart logic
+  cartLogic() {
+    cartContent.addEventListener("click", (event) => {
+      //For the remove button in the cart
+      if (event.target.classList.contains("remove-item")) {
+        let removeItem = event.target;
+        let id = removeItem.dataset.id;
+        cartContent.removeChild(removeItem.parentElement.parentElement);
+        this.removeItem(id);
+      }
+      //for the increament of the variant
+      else if (event.target.classList.contains("fa-chevron-up")) {
+        let increaseNumber = event.target;
+        let id = increaseNumber.dataset.id;
+        let tempItem = cart.find((item) => item.id === id);
+        tempItem.amount = tempItem.amount + 1;
+        Storage.saveCart(cart);
+        this.setCartValues(cart);
+        increaseNumber.nextElementSibling.innerText = tempItem.amount;
+      }
+      //for the reduction of the variant
+      else if (event.target.classList.contains("fa-chevron-down")) {
+        let lowerAmount = event.target;
+        let id = lowerAmount.dataset.id;
+        let tempItem = cart.find((item) => item.id === id);
+        tempItem.amount = tempItem.amount - 1;
+        if (tempItem.amount > 0) {
+          Storage.saveCart(cart);
+          this.setCartValues(cart);
+          lowerAmount.previousElementSibling.innerText = tempItem.amount;
+        } else {
+          cartContent.removeChild(lowerAmount.parentElement.parentElement);
```

Comment:
we have alot going on here too, please let them in functions. and there is some logic that can be reused also.

Reply:
Ok then.. I'll create a function for them